Firecracker CNI

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/890
<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
This adds CNI configs and the tc-redirect-tap CNI plugin dependency. Sadly this doesn't appear to be usable on ChromeOS due to missing kernel modules, but this will still be useful when running this on AWS or other distros.
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
Both bash scripts were run successfully
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Set up basic CNI networking stuff

While this isn't working in ChromeOS, we're still going to need a way
to manage CNI networking in AWS or on other platforms. This commit
adds the current progress towards getting CNI networking used.

Commit:
Add tc-redirect-tap CNI plugin

Even though chromeos doesn't have the necessary kernel module, we
should still install this for completeness. It should also be easy to
port to other distros

twunderlich-grapl commented on etc/chromeos/lib/installs.sh:
```diff
@@ -322,3 +368,20 @@ install_sqlx_prepare_deps() {
     sudo apt install --yes netcat # used for `nc`
 }
+
+install_tc_redirect_tap() {
```
Since this requires golang, I'm not making this mandatory. 
I'm split on whether it makes sense to auto-install golang or just package this up somewhere for downloading in the future

christophermaier replied:
Packaging it and downloading from, say, Cloudsmith probably makes the most sense long-term.

Short-term, you could compile it in a Go container and just spit the compiled binary out.

Commit:
Download the tc-redirect-tap plugin from cloudsmith

Instead of compiling this from source which requires golang installed
and set up properly, we're just throwing this into cloudsmith for the
moment and downloading it directly from there. Long-term, either
the project will create github releases or we can fork and create the
releases ourselves. However, this changes rarely (4 merged PRs in all
of 2021), so we can update this manually ourselves if necessary
